### ISSUE TITLE
problem in alias path.

### ISSUE THREAD

**Issue by vrwebdesign** [adonisjs/adonis-framework#845]
when I define alias like this 
```
const aliases = {
  MyModel: path.resolve(__dirname, "../app", "helper", "MyModel")
}
```
everything work but on the server I get an Error 

> Error: Cannot find module '/home/development/www/Cashineh-APIadonis/node_modules/@adonisjs/fold/src/Ioc/home/development/www/Cashineh-APIadonis/app/helper/MyModel'

I Change my alias path method to this one

```
const aliases = {
  MyModel: "App/helper/MyModel"
}
```
this worked but my hook not work any more.
whats the problem?

**Comment by thetutlage** [comment]
Currently aliases only work for IoC container bindings created via `Ioc.bind`. I think it should work for normal namespaces too like `App/Models/MyModel`. For that, I will have to add support for that.

> this worked but my hook not work any more.

This method worked, because you defined an absolute path, so IoC container was able to resolve this.

In order for hooks to work, Lucid model uses IoC container lifecycle methods to boot the model behind the scenes. Since your aliases was not a normal IoC container binding, but instead a full path, those lifecycle methods were never called.

In short, I will add support for define aliases for autoloaded directories too. Which means you will be able to do.

```js
const aliases = {
  MyModel: 'App/Models/MyModel'
}
```

But again defining an absolute path for an alias, will never trigger the IoC container lifecycle hooks

**Comment by vrwebdesign** [comment]
MyModel.js
```
const Model = use("Model");
class MyModel extends Model {

  static listOption(qs) {
      ...  
  }
}
```

this is my model 
what can I do to be a normal IoC container?

**Comment by thetutlage** [comment]
Can you share a sample repo with what exactly you are trying to do?

**Comment by vrwebdesign** [comment]
```
  static listOption(qs) {
    let { filters, page, perPage, sort, withArray } = qs;

    filters = (filters && JSON.parse(filters)) || [];
    if (!JSON.stringify(filters).includes("isDeleted")) {
      filters.push("isDeleted:0:=");
    }
    page = parseInt(page) || 1;
    perPage = parseInt(perPage) || 10;
    sort = sort && sort.split("-");
    let orderby_direction, orderby_field;
    if (sort && sort.length === 2) {
      orderby_direction = "DESC";
      orderby_field = sort[1];
    } else if (sort && sort.length === 1) {
      orderby_field = sort[0];
    } else {
      orderby_field = "updated_at";
      orderby_direction = "DESC";
    }

    let query = super.query();
    if (withArray && withArray.length) {
      withArray.forEach(name => {
        query = query.with(name);
      });
    }
    for (let filter of filters) {
      let [property, value, opt] = filter.split(":");
      if (opt === "like") value = `%${value}%`;
      if (property.includes(".")) {
        let [a, b] = property.split(".");
        if (withArray && withArray.indexOf(a) !== -1) {
          query = query.whereHas(a, builder => {
            builder.where(b, opt || "=", value);
          });
          continue;
        }
      }
      query = query.where(property, opt || "=", value);
    }
    query = query
      .orderBy(orderby_field, orderby_direction)
      .paginate(page, perPage);
    return query;
  }
```
I have a listOption method that query in table and I want to use this method on all of my table and use result in a dataGrid.

**Comment by thetutlage** [comment]
The original question was path aliases not working and what you have posted now is no where near that I can use to reproduce the actual issue.

**Comment by vrwebdesign** [comment]
When I use absolute path my hook not work
I want a way to add my custom function to all of my model

**Comment by thetutlage** [comment]
I asked you to share a sample repo https://github.com/adonisjs/adonis-framework/issues/845#issuecomment-386509269#issuecomment-385377860, which can help me reproduce this issue. Please do not share your entire codebase.  Just a sample repo with the minimum amount of code to reproduce the issue.

**Comment by vrwebdesign** [comment]
when use this my hook not work

**Comment by vrwebdesign** [comment]
use path.resolve work on widows system but not work in the linux
I don't know why

**Comment by thetutlage** [comment]
It works fine for me. Also your directory is uppercase `Helpers` and you are using lowercase `helpers`

**Comment by vrwebdesign** [comment]
in my environment (windows 8) hook not work

**Comment by thetutlage** [comment]
Where are you adding the hook to the model? https://github.com/vrwebdesign/adonis-path/blob/master/app/Helper/MyModel.js

**Comment by vrwebdesign** [comment]
https://github.com/vrwebdesign/adonis-path/blob/master/app/Models/Order.js
in this file i add a afterCreate hook
but not work

**Comment by vrwebdesign** [comment]
https://github.com/adonisjs/adonis-lucid/issues/341#issuecomment-386861317
